src/app/portlets/content-types/fields/dot-content-type-fields-variables/dot-content-type-fields-variables.component.ts

```diff
@@ -0,0 +1,221 @@
+import {
+    Component,
+    Input,
+    SimpleChanges,
+    OnChanges,
+    OnInit,
+    OnDestroy,
+    ViewChild
+} from '@angular/core';
+import { DotMessageService } from '../../../../api/services/dot-messages-service';
+import {
+    DotFieldVariablesService,
+    DotFieldVariableParams
+} from '../service/dot-field-variables.service';
+import { DotHttpErrorManagerService } from '../../../../api/services/dot-http-error-manager/dot-http-error-manager.service';
+import { ResponseView } from 'dotcms-js';
+import { take, takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs/internal/Subject';
+import { Table } from 'primeng/table';
+import * as _ from 'lodash';
+
+export interface FieldVariable {
+    id?: string;
+    clazz?: string;
+    fieldId?: string;
+    key: string;
+    value: string;
+}
+
+@Component({
+    selector: 'dot-content-type-fields-variables',
+    styleUrls: ['./dot-content-type-fields-variables.component.scss'],
+    templateUrl: './dot-content-type-fields-variables.component.html'
+})
+export class DotContentTypeFieldsVariablesComponent implements OnInit, OnChanges, OnDestroy {
+    @ViewChild('table')
+    table: Table;
+
+    @Input()
+    field: DotFieldVariableParams;
+
+    fieldVariables: FieldVariable[] = [];
+    fieldVariablesBackup: FieldVariable[] = [];
+    messages: { [key: string]: string } = {};
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        public dotMessageService: DotMessageService,
+        private fieldVariablesService: DotFieldVariablesService
+    ) {}
+
+    ngOnInit(): void {
+        this.dotMessageService
+            .getMessages([
+                'contenttypes.field.variables.key_header.label',
+                'contenttypes.field.variables.value_header.label',
+                'contenttypes.field.variables.actions_header.label',
+                'contenttypes.field.variables.value_no_rows.label'
+            ])
+            .pipe(take(1))
+            .subscribe((messages: { [key: string]: string }) => {
+                this.messages = messages;
+                this.initTableData();
+            });
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.field.currentValue && !changes.field.firstChange) {
+            this.initTableData();
+        }
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Handle Delete event
+     * @param {FieldVariable} fieldVariable
+     * @param {number} fieldIndex
+     * @memberof DotContentTypeFieldsVariablesComponent
+     */
```

Comment:
I think this comment is wrong, maybe is better something like: "delete a Field Variable"

Also maybe we should start to write a little more detail in our comments, something like:

'...if the FieldVariable already has a id then a request is sent to the end point ...'